lib/widgets/action_sheet.dart

```diff
@@ -402,3 +405,70 @@ class ShareButton extends MessageActionSheetMenuItemButton {
     }
   }
 }
+
+class MarkAsUnreadButton extends MessageActionSheetMenuItemButton {
+  MarkAsUnreadButton({
+    super.key,
+    required super.message,
+    required super.messageListContext,
+  }) : narrow = MessageListPage.ancestorOf(messageListContext).narrow;
+
+  final Narrow narrow;
+
+  @override IconData get icon => Icons.mark_chat_unread_outlined;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return zulipLocalizations.actionSheetMarkAsUnread;
+  }
+
+  @override void onPressed(BuildContext context) async {
+    Navigator.of(context).pop();
+    final zulipLocalizations = ZulipLocalizations.of(messageListContext);
+
+    try {
+      final store = PerAccountStoreWidget.of(messageListContext);
+      final connection = store.connection;
+      Anchor anchor = NumericAnchor(message.id);
+      final apiNarrow = narrow.apiEncode();
+
+      while (true) {
+        final result = await updateMessageFlagsForNarrow(connection,
+          anchor: anchor,
+          includeAnchor: true,
+          numBefore: 0,
+          numAfter: 1000,
```

Comment:
Seems like we have [5000](https://github.com/zulip/zulip-mobile/blob/e352f563ecf2fa9b09b688d5a65b6bc89b0358bc/src/action-sheets/index.js#L203-L280) from zulip-mobile, and [1000-1](https://github.com/zulip/zulip/blob/43e1a643c0321b351d1f091f02eacc95caafb796/web/src/unread_ops.ts#L250) from web. How did you pick this number?